### PR TITLE
Move mixer client integration tests from istio/proxy  repo

### PR DESCRIPTION
**Pull request by qiwzhang**
So that istio/proxy repo doesn't have to deal with golang.  It will be a pure c++ repo.

Added a script to run it sequentially.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -260,11 +260,15 @@ security-test:
...
+.PHONY: mixer-client-test
+mixer-client-test:
+	mixer/test/client/run_test.sh
+
 common-test:
 	go test ${T} ./pkg/...
 
 # Run coverage tests
-go-test: pilot-test mixer-test security-test broker-test common-test
+go-test: pilot-test mixer-test security-test broker-test common-test mixer-client-test
```
Comment (rshriram) [inline Makefile]: why do we need to run this as part of our full tests?

**Comment by rshriram** [comment]
In general, i think this approach will lead to trouble, because any feature update to mixer filter will need two PRs to be coordinated across istio/proxy and istio/istio. Or features will get into istio/proxy without tests, and we run the risk of temporary instability until the corresponding tests land in istio/istio and are successful.

**Comment by kyessenov** [comment]
This is why we should use branches for breaking changes. We can test
against branches and then merge together in both repos.

File: Makefile [hunk 2]
```diff
@@ -260,11 +260,15 @@ security-test:
 	go test ${T} ./security/pkg/...
 	go test ${T} ./security/cmd/...
 
+.PHONY: mixer-client-test
+mixer-client-test:
+	mixer/test/client/run_test.sh
```
Comment (mattdelco) [inline Makefile]: Offhand it seems like the 1-line functionality of run_test.sh should be a good candidate to instead implement as a rule within make.

**Comment by lizan** [comment]
I agree with @rshriram this might lead trouble if features get into istio/proxy, though I think we should have at least some basic unit test and functional test, at least without depending on mixer server go package, in C++ level. That would also benefit the potential upstreaming mixer filter work.

**Comment by kyessenov** [comment]
Yes, it's only meant to encourage good practices. Self-contained tests in
proxy are important for upstreaming, and breaking changes are harmful for
production readiness. New non-breaking changes to mixer filter can be gated
by feature flags that default to `false`.

File: Makefile (hunk 2 above)
Comment (qiwzhang) [reply Makefile]: try to make "go test ./mixer/.." works for the new tests: ( need to support running the tests in parallel).  Once it works, this rule will be removed

**Comment by qiwzhang** [comment]
The focus of this PR is to remove GO in istio/proxy.  It is very painful to build Envoy and go Mixer in one repo.   
Yes, eventually, we need c++ unit-test for mixer filter codes in proxy.  Before that, we have to rely on the go testing code in istio.

**Comment by qiwzhang** [comment]
@lizan @mandarjog The PR is ready for review.  PTAL.  Thanks

**Comment by qiwzhang** [comment]
Hi @mandarjog 

Did you fix mixer-ratelimit test?  If no, why e2e-smoke is marked as "required"

e2e-smoke.sh  failed at

FAIL: TestRateLimit

Beside,  this PR only add some go tests,  I don't think it will alter e2e-smoke tests.  It only adds some tests to "make mixer-test"  for presubmit

**Comment by ldemailly** [comment]
it's fixed now (rate limit test)

/test e2e-smoke

**Comment by ldemailly** [comment]
should this be an import (like when we moved say mixer to istio) to preserve the git history ?

**Comment by mandarjog** [comment]
It is non trivial to just move some parts of another repo, and these are just tests.

**Pull request merged**